imap_data_access/processing_input.py

```diff
@@ -122,7 +126,8 @@ def _set_attributes_from_filenames(self):
             else:
                 data_type.add(self.input_type.value)
             descriptor.add(path_validator.descriptor)
-            file_path_list.append(str(path_validator.filename))
+            # Store download path. This gets used in `imap_processing` cli
+            file_path_list.append(path_validator.construct_path())
```

Comment:
as Luisa said: 
```suggestion
            imap_file_paths.append(path_validator)
```

Here, path_validator will be `ScienceFilePath` or `AncillaryFilePath` which are both `ImapFilePath` types, so that should work.

Reply:
I can't comment on it, but then also in ln 148 you would need to use `self.imap_file_paths.get_filename()`. These changes mean we're storing `ScienceFilePath` objects with all those convenient functions.